src/components/shared/StarsRating.js

```diff
@@ -1,34 +1,144 @@
-import Star from "@material-ui/icons/StarBorder";
-import React from "react";
+import Star from "@material-ui/icons/StarRate";
+import React, { Component } from "react";
 import { withStyles } from "@material-ui/core";
 import PropTypes from "prop-types";
+import { connect } from "react-redux";
+
+import {
+  setNewRatingForSong,
+  loadSongsRating
+} from "../../store/actionCreators/rating";
 
 const styles = {
+  starsRatingContainer: {
+    display: "flex"
+  },
   starsRating: {
     color: "gray",
-    "&:hover": {
-      color: "orange"
-    }
+    overflow: "hidden",
+    cursor: "pointer"
+  },
+  active: {
+    color: "orange"
   }
 };
 
-class StarsRating extends React.Component {
+class StarsRating extends Component {
   static propTypes = {
-    classes: PropTypes.object
+    classes: PropTypes.object,
+    loadSongsRating: PropTypes.func,
+    setNewRatingForSong: PropTypes.func,
+    songId: PropTypes.string,
+    userUid: PropTypes.string,
+    rating: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
+    maxRating: PropTypes.number
   };
 
-  render() {
+  state = {
+    rating: this.getSongRating,
+    tempRating: null
+  };
+
+  handleMouseover = newRating => {
+    this.setState({
+      tempRating: newRating
+    });
+  };
+
+  handleMouseout = () => {
+    this.setState({
+      tempRating: null
+    });
+  };
+
+  getSongsRating = () => {
+    return this.props.loadSongsRating(this.props.userUid);
+  };
+
+  componentDidMount() {
+    this.getSongsRating();
+  }
+
+  getSongRating = () => {
+    const ratingObj = this.props.rating;
+    const id = this.props.songId;
+    return ratingObj[id];
+  };
+
+  rate = rating => {
+    this.setState({
+      ...this.state,
+      rating: rating + 1,
+      tempRating: null
+    });
+    this.props.setNewRatingForSong(
+      this.props.userUid,
+      this.props.songId,
+      rating + 1
+    );
+  };
+
+  componentWillUpdate(nextProps, nextState) {
+    if (nextProps !== this.props) {
+      this.setState(() => ({
+        ...this.state,
+        rating: this.getSongRating()
+      }));
+    }
+  }
+
+  createStarsMaxCount = count => {
+    return new Array(count).fill(0).map((_, i) => i);
+  };
+
+  createStars = count => {
     const { classes } = this.props;
 
+    return this.createStarsMaxCount(count).map((item, i) => {
+      let starClass = classes.starsRating;
+      if (
+        (this.state.tempRating >= i && this.state.tempRating !== null) ||
+        (this.state.rating >= i + 1 && this.state.rating !== null)
+      ) {
+        starClass = `${classes.starsRating} ${classes.active}`;
+      }
```

Comment:
please extract this code for computing className into method `getStartClassName`. Then you can use it directly inside jsx code in render. And you won't need `createStars` method